packages/react-router/src/components/FocusContextProvider/FocusContextProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import React, {useRef, useEffect, ReactNode} from 'react';
+import {FocusContext} from '../../context';
+
+export function FocusContextProvider({children}: {children: ReactNode}) {
+  const focusRef = React.createRef<HTMLElement>();
+  const focus = () => {
+    return focusRef?.current?.focus();
+  };
+
+  const firstRender = useRef(true);
+
+  useEffect(() => {
+    if (!firstRender.current) {
+      focus();
+    }
+  });
+
+  useEffect(() => {
+    firstRender.current = false;
+  }, []);
```

Comment:
We're keeping track of the first render here because on the first load of the page, we _should not_ force the focus (we should let the browser do whatever native behaviour it wants). Once the user starts navigating through the SPA, only then should we start forcing focus somewhere.